src/app/components/registration/registration.component.ts

```diff
@@ -0,0 +1,53 @@
+
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UserService } from 'src/app/services/user.service';
+
+
+@Component({
+  selector: 'app-registration',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.scss']
+})
+export class RegistrationComponent implements OnInit {
+  registrationForm: FormGroup;
+
+  constructor(private formBuilder: FormBuilder, private userService: UserService, private router: Router) {
+    this.registrationForm = this.formBuilder.group({
+      name: ['', Validators.required],
+      email: ['', [Validators.required, Validators.email]],
+      password: ['', Validators.required],
+      confirmPassword: ['', Validators.required]
+    });
```

Comment:
I think it is not necessary to initialize the `registrationForm` inside the constructor. If you don't use values, that comes from dependencies in constructor, it is better to declare the `registrationForm` variable in line 14, together with initialization:

```suggestion
  registrationForm = this.formBuilder.group({
      name: ['', Validators.required],
      email: ['', [Validators.required, Validators.email]],
      password: ['', Validators.required],
      confirmPassword: ['', Validators.required]
    });
    
  constructor(private formBuilder: FormBuilder, private userService: UserService, private router: Router) {}
```